pkg/anypoint/types.go

```diff
@@ -221,5 +221,93 @@ type AnalyticsEvent struct {
+const (
+	ExternalOauth       = "external-oauth2-access-token-enforcement"
+	ClientID            = "client-id-enforcement"
+	SlaAuth             = "sla-based"
+	ApiKey              = "apiKey"
+	Oauth2              = "oauth2"
+	TokenUrl            = "tokenUrl"
+	Scopes              = "scopes"
+	Header              = "header"
+	AccessCode          = "accessCode"
+	Authorization       = "authorization"
+	CredOrigin          = "credentialsOriginHasHttpBasicAuthenticationHeader"
+	DescClienCred       = "Provided as: client_id:<INSERT_VALID_CLIENTID_HERE> \n\n client_secret:<INSERT_VALID_SECRET_HERE>\n\n"
+	DescOauth2          = "This API supports OAuth 2.0 for authenticating all API requests"
+	ErrAuthNotSupported = "only Oauth and ClientID enforcement(ApiKey) security policies are supported"
+	ErrSpecNotSupported = "Invalid Spec Type, Only OAS specs are supported"
+	ClientIDProp        = "client_id"
+	ClientSecretProp    = "client_secret"
+	AppName             = "appName"
+	Description         = "description"
+	TierLabel           = "SLA Tier"
+)
+
+type Application struct {
+	Name         string `json:"name"`
+	Description  string `json:"description"`
+	ClientId     string `json:"clientId"`
+	ClientSecret string `json:"clientSecret"`
+	Id           int64  `json:"id"`
+	APIEndpoints bool   `json:"apiEndpoints"`
+}
+
+type AppRequestBody struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+type Contract struct {
+	APIID           string `json:"apiId"`
+	EnvironmentId   string `json:"environmentId"`
+	AcceptedTerms   bool   `json:"acceptedTerms"`
+	OrganizationId  string `json:"organizationId"`
+	GroupId         string `json:"groupId"`
+	AssetId         string `json:"assetId"`
+	Version         string `json:"version"`
+	VersionGroup    string `json:"versionGroup"`
+	RequestedTierID int64  `json:"requestedTierId,omitempty"`
+}
+
+type ContractResponse struct {
+	Id            int    `json:"id"`
+	Status        string `json:"status"`
+	ApplicationId int    `json:"applicationId"`
+	ClientId      string `json:"clientId"`
+	ClientSecret  string `json:"clientSecret"`
+	API           API    `json:"api"`
+}
+
+type Tiers struct {
+	Total int       `json:"total"`
+	Tiers []SLATier `json:"tiers"`
+}
+
+type SLATier struct {
+	Id          int         `json:"id"`
+	Name        string      `json:"name"`
+	Description interface{} `json:"description"`
+	Limits      []Limits    `json:"limits"`
+	Status      string      `json:"status"`
+}
+
+type Limits struct {
+	Visible                  bool `json:"visible"`
+	TimePeriodInMilliseconds int  `json:"timePeriodInMilliseconds"`
+	MaximumRequests          int  `json:"maximumRequests"`
+}
+
+// errorString is a trivial implementation of error.
+type errorString struct {
+	s string
+}
+
+func (e *errorString) Error() string {
+	return e.s
+}
+
+// New returns an error that formats as the given text.
+func New(text string) error {
+	return &errorString{text}
+}
```

Comment:
Look at the errors package in the SDK, to create predefined errors with error codes